iotedgedev/modules.py

```diff
@@ -27,86 +31,48 @@ def build(self):
+                        docker_file = module_json.get_platform_by_key(arch)
 
                         self.output.info(
                             "PROCESSING DOCKER FILE: " + docker_file)
 
                         docker_file_name = os.path.basename(docker_file)
-
-                        # assume /Docker/{runtime}/Dockerfile folder structure
-                        # image name will be the same as the module folder name, filter-module
-                        # tag will be {runtime}{ext}{container_tag}, i.e. linux-x64-debug-jong
-                        # runtime is the Dockerfile immediate parent folder name
-                        # ext is Dockerfile extension for example with Dockerfile.debug, debug is the mod
-                        # CONTAINER_TAG is env var
-
-                        # i.e. when found: filter-module/Docker/linux-x64/Dockerfile.debug and CONTAINER_TAG = jong
-                        # we'll get: filtermodule:linux-x64-debug-jong
-
-                        runtime = os.path.basename(
-                            os.path.dirname(docker_file))
-                        ext = "" if os.path.splitext(docker_file)[
-                            1] == "" else "-" + os.path.splitext(docker_file)[1][1:]
                         container_tag = "" if self.envvars.CONTAINER_TAG == "" else "-" + \
                             self.envvars.CONTAINER_TAG
 
-                        tag_name = runtime + ext + container_tag
+                        tag_name = module_json.tag_version + container_tag
 
-                        # construct the build output path
-                        build_path = os.path.join(
-                            os.getcwd(), "build", "modules", module, runtime)
-                        if not os.path.exists(build_path):
-                            os.makedirs(build_path)
-
-                        # dotnet publish
+                        # publish module
                         self.output.info(
-                            "PUBLISHING PROJECT: " + project_files[0])
-
-                        self.utility.exe_proc(["dotnet", "publish", project_files[0], "-f", "netcoreapp2.0",
-                                               "-o", build_path, "-v", self.envvars.DOTNET_VERBOSITY])
+                            "PUBLISHING PROJECT: " + module_dir)
 
-                        # copy Dockerfile to publish dir
-                        build_dockerfile = os.path.join(
-                            build_path, docker_file_name)
-
-                        copyfile(docker_file, build_dockerfile)
-
-                        image_destination_name = "{0}/{1}:{2}".format(
-                            self.envvars.CONTAINER_REGISTRY_SERVER, module, tag_name).lower()
+                        mod_proc.publish(module_dir)
+                        image_destination_name = "{0}/{1}:{2}-{3}".format(
+                            self.envvars.CONTAINER_REGISTRY_SERVER, module, tag_name, docker_file_name).lower()
```

Comment:
Won't this be filtermodule:0.0.0-Dockerfile?  I think you mean "arch" here and not docker_file_name